### PR TITLE
Fail a test on pre-provisioned Cinder volume deletion error

### PR DESCRIPTION
**Pull request by jsafrane**
/kind cleanup
**What this PR does / why we need it**:

We see number of leaked Cinder volumes in a downstream test suite. While we're debugging this, we noticed a few issues:

1. The tests should log error output of each failed `cinder delete <xyz>`.
2. The tests should use volume ID instead of volume name in `cinder delete <xyz>`, since multiple CI jobs can run in a single OpenStack project and accidentally choose the same volume name.
3. The tests should be marked as failed when pre-provisioned volume deletion fails.

In addition, I added `cinder show <xyz>` output when something fails, to be able to diagnose volume health from Cinder point of view.

This PR does not fix the leak, that one is still under investigation.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by jsafrane** [comment]
cc @kubernetes/cloud-provider-openstack-maintainers for review

**Comment by jsafrane** [comment]
I don't see any e2e openstack job in the list.

File: test/e2e/storage/drivers/in_tree.go [hunk 1]
```diff
@@ -1199,16 +1200,23 @@ func (v *cinderVolume) DeleteVolume() {
...
 			return
 		}
-		framework.Logf("Failed to delete volume %s: %v", name, err)
+		framework.Logf("Failed to delete volume %s / %s: %v\n%s", id, name, err, string(output))
 	}
-	framework.Logf("Giving up deleting volume %s: %v\n%s", name, err, string(output[:]))
+	// Timed out, try to get "cinder show <volume>" output for easier debugging
+	showOutput, showErr := exec.Command("cinder", "show", id).CombinedOutput()
+	if showErr != nil {
+		framework.Logf("Failed to show volume %s / %s: %v\n%s", id, name, showErr, string(showOutput))
```
[the remark this answers is not in the document]
Comment (jsafrane) [reply test/e2e/storage/drivers/in_tree.go:1215]: What I noticed in our (OpenShift) env is that `cinder delete` failed with

```
Invalid volume: Volume status must be available or error or [...]
```

And `cinder show` showed the volume is in some bad state.

File: test/e2e/storage/drivers/in_tree.go [hunk 2]
```diff
@@ -1199,16 +1200,23 @@ func (v *cinderVolume) DeleteVolume() {
...
+	} else {
+		framework.Logf("Volume %s / %s:\n%s", id, name, string(showOutput))
+	}
+	framework.Failf("Failed to delete pre-provisioned volume %s / %s: %v\n%s", id, name, err, string(output[:]))
```
[the remark this answers is not in the document]
Comment (jsafrane) [reply test/e2e/storage/drivers/in_tree.go:1219]: `Failf` makes this error recorded as the reason why a test failed. See a random test failure here: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/94881/pull-kubernetes-e2e-gce-csi-serial/1307100981535707136

If you click on a failed test, it shows nice output:
```
test/e2e/storage/testsuites/volume_expand.go:238
Sep 19 00:13:41.897: While creating pods for resizing
Unexpected error:
    <*errors.errorString | 0xc002ba1130>: {
        s: "pod \"pod-7ee298ca-bc8a-4b19-b0fd-0ae251e22924\" is not Running: timed out waiting for the condition",
    }
    pod "pod-7ee298ca-bc8a-4b19-b0fd-0ae251e22924" is not Running: timed out waiting for the condition
occurred
test/e2e/storage/testsuites/volume_expand.go:255
```

And with `Failf` here, we get stdout/stderr in that UI.

File: test/e2e/storage/drivers/in_tree.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsafrane) [reply test/e2e/storage/drivers/in_tree.go:1215]: Sorry, re-reading your comment again, if `cinder show` returns error, it probably won't show volume status, still, showing something like `Invalid filters all_tenants,name are found in query options. (HTTP 400)` is still more useful than plain `exit status 1`

**Comment by ramineni** [comment]
looks good to me. Thanks

File: test/e2e/storage/drivers/in_tree.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (oomichi) [reply test/e2e/storage/drivers/in_tree.go:1219]: Ah, I see the point.
Thanks for your explanation.

/lgtm

**Pull request merged**